### ISSUE TITLE
Scheduled query executiosn fail when a query uses dynamic date range parameters

### ISSUE THREAD

**Issue by susodapop** [getredash/redash#4514]
### Issue Summary

Since dynamic date ranges are evaluated in the front-end, scheduled executions fail without warning or notice.

### Steps to Reproduce

1. Write a query with a date range parameter
2. Configure the date range parameter to use a dynamic value
3. Execute the query and note its last execution time
4. Schedule the query to execute within a reasonable time (60 seconds e.g.)
5. After the time allotted, observe that the last executed time has not updated

I expect the interface to warn about this behavior. Ideally, we can perform the calculation in the back-end so that scheduled executions can work.

### Technical details:

* Redash Version: v9 alpha
* Browser/OS: FF / Chrome / Safari
* How did you install Redash: SaaS

**Comment by wtfzambo** [comment]
Yup, encountered this problem just today. Using a self hosted version of redash.
Any idea of a potential workaround  that I can easily implement to circumvent this?

**Comment by susodapop** [comment]
My workaround uses the API. The example [here](https://github.com/getredash/redash-toolbelt/blob/master/redash_toolbelt/examples/refresh_dashboard.py) does it, including dynamic dates.

**Comment by wtfzambo** [comment]
@susodapop 
Hi, thanks for the reply. I've checked the repo you linked but I'm a bit lost.

Could you elaborate a bit more in depth how to implement the toolbelt? I'm working on a self hosted instance but the selfhosting itself was made by my company's developer, so I have zero clue what that implies.

**Comment by susodapop** [comment]
@wtfzambo I'm fixing a couple issues with that script. Will follow-up with you once completed.

**Comment by susodapop** [comment]
@wtfzambo I made an example using `refresh_dashboard.py`. It updates a whole dashboard of queries.

But you can borrow the logic to make a single query updater or whatever you need. This _doesn't_ require you to have admin access to the Redash instance. All you need is your API key (as shown in the video).

[![](https://img.youtube.com/vi/-7CpHW0D1J8/0.jpg)](http://www.youtube.com/watch?v=-7CpHW0D1J8} "Click to play on Youtube.com")

**Comment by wtfzambo** [comment]
@susodapop yo, thanks a lot for taking the time to make this video, it was really explanatory.
I am actually admin, but I thought I needed to edit some things in the docker and I don't have access to that (nor would I know tf to do anyway).

So, given that I host my redash inside a VPC in AWS, I guess to make this run on a daily basis I need to make a lambda function inside the same VPC and schedule a cloudwatch event to trigger said lambda. Would that work?

**Comment by susodapop** [comment]
@wtfzambo You _could_ do that. I usually just add a cron job.

**Comment by wtfzambo** [comment]
@susodapop a cron job where? in the redash instance itself?

**Comment by susodapop** [comment]
@wtfzambo You can run the cron job on any box that has network access to your Redash instance. It can be your laptop, a raspberry pi, a docker container on your NAS or server, a VPS somewhere (kinda like your cloudwatch instance).

I usually run them on a raspberry pi. Entirely up to you.

You probably _wouldn't_ run it in the Redash Docker container though, since updates down the road might blow away your settings.

**Comment by wtfzambo** [comment]
@susodapop aaaah ok! Sorry for the stupid question, I was confused at first because since I'm operating for a business, I would never have considered running a cron on my personal laptop :D
